crypto/ffc/ffc_params.c

```diff
@@ -27,11 +27,19 @@ void ossl_ffc_params_init(FFC_PARAMS *params)
 
 void ossl_ffc_params_cleanup(FFC_PARAMS *params)
 {
+#ifndef FIPS_MODULE
     BN_free(params->p);
     BN_free(params->q);
     BN_free(params->g);
     BN_free(params->j);
     OPENSSL_free(params->seed);
```

Comment:
Given the significant amount of `#ifndef FIPS_MODULE` and near-duplication of code, I wonder if it would be better to introduce a `BN_free_sensitive()` and `OPENSSL_free_sensitive()` which become `xxx_clear_free()` for FIPS otherwise just `xxx_free()`.

But since we already have and use `xxx_clear_free()` under what circumstances is it best to only clear sensitive data in the FIPS module?

i.e. should these changes use `xxx_clear_free()` unconditionally?

Reply:
See other PRs which discussed this and linked to this one.

Currently for default provider critical security parameters (e.g. private key) are zeroised, but not public security parameters (e.g. public salt).

If we were to follow your suggestion and want to still support above combinations we would need `clear_critical`, `clear_public` as separate ones. Rather than catch both `clear_sensitive`.

I am worried about adding new symbols to the library API and ABI. Maybe the above can be defines only, or weak symbol aliases.

Reply:
Sorry (again) by `sensitive` I meant "what we want to clear in FIPS but not in non-FIPS" i.e. public. There must be a better name than `xxx_free_sensitive()` to imply "maybe clear; always free". We do just need two more APIs as the existing `xxx_clear_free()` ones are fine for the private values that must always be cleared.